### PR TITLE
Votes items should be sorted by time and hotness

### PR DESCRIPTION
**Pull request by technicalcapt**
#63

### CODE REVIEW

**Comment by technicalcapt** [comment]
`

    order = :math.log10(max(abs(s), 1))
    sign =
      cond do
        s > 0 -> 1
        s < 0 -> -1
        true -> 0
      end
    seconds = epoch_seconds(naive_datetime) - 1134028003 # time of 7:46:43 am December 8, 2005
    Float.round(sign * order + seconds / 45000, 7)

`
I'm taking the time of 7:46:43 am December, 8, 2005 in the example.
Help me with this timeline!

File: mix.lock [hunk 1]
```diff
@@ -1,5 +1,6 @@
 %{"bunt": {:hex, :bunt, "0.2.0", "951c6e801e8b1d2cbe58ebbd3e616a869061ddadcc4863d0a2182541acae9a38", [:mix], []},
   "certifi": {:hex, :certifi, "1.0.0", "1c787a85b1855ba354f0b8920392c19aa1d06b0ee1362f9141279620a5be2039", [:rebar3], []},
+  "combine": {:hex, :combine, "0.9.6", "8d1034a127d4cbf6924c8a5010d3534d958085575fa4d9b878f200d79ac78335", [:mix], []},
```
Comment (awwong1) [inline mix.lock]: why is this added?

Comment (technicalcapt) [inline mix.lock]: That maybe timex package that i haven't cleared!

Comment (awwong1) [inline mix.lock]: yeah, not sure if timex is necessary, don't see you using it anywhere

Comment (technicalcapt) [inline mix.lock]: Yeah i don't use it. Let me remove it!

File: lib/udia/helpers/helper.ex [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      end
+
+    order = :math.log10(max(abs(s), 1))
+    sign =
+      cond do
+        s > 0 -> 1
+        s < 0 -> -1
+        true -> 0
+      end
+    seconds = epoch_seconds(naive_datetime) - 1134028003 # time of 7:46:43 am December 8, 2005
```
Comment (awwong1) [inline lib/udia/helpers/helper.ex]: I'm pretty sure this is supposed to be the current timestamp

Comment (technicalcapt) [inline lib/udia/helpers/helper.ex]: You're right. But at the moment, time when create a post is naive datetime (without timezone). I think this should be fixed

Comment (awwong1) [inline lib/udia/helpers/helper.ex]: I thought`timestamps()` stores everything in UTC time. Why would this break anything? All calculation is performed using UTC time on the server and will be rendered to a user's local time on client

Comment (technicalcapt) [inline lib/udia/helpers/helper.ex]: Timestamps opts defaults
`
:type - column type, defaults to :naive_datetime
`
 This should be changed to `:utc_datetime`

Comment (awwong1) [inline lib/udia/helpers/helper.ex]: 💩  Let's add that in and set that as default across all models. Can you add that in this PR and perform the migrations (if any are needed?)

Comment (technicalcapt) [inline lib/udia/helpers/helper.ex]: Are you going to drop the db again? lol

Comment (awwong1) [inline lib/udia/helpers/helper.ex]: only if I have to

Comment (technicalcapt) [inline lib/udia/helpers/helper.ex]: Ok, done!
